File: Zig/start-code/startcode.zig
const std = @import("std");

pub fn main() void {
    // Begrüßung
    std.debug.print("Willkommen zu meinem Zig-Programm!\n", .{});

    // Nutzereingabe
    std.debug.print("Bitte geben Sie Ihren Namen ein: ");
    var name = try std.io.getStdIn().readLineAlloc(std.heap.page_allocator);

    // Begrüßung mit dem Namen
    std.debug.print("Hallo, {}! Schön, dass Sie hier sind.\n", .{name});
    std.debug.print("Das Programm wurde beendet.\n", .{});
}

File: Rust/start-code/startcode.rs
// Beschreibung: Ein einfaches Rust-Programm
// Datum: 25.01.2025

use std::io;

fn main() {
    // Begrüßung
    println!("Willkommen zu meinem Rust-Programm!");

    // Nutzereingabe
    println!("Bitte geben Sie Ihren Namen ein:");

    let mut name = String::new();
    io::stdin()
        .read_line(&mut name)
        .expect("Fehler beim Lesen der Eingabe");

    // Entfernen des Zeilenumbruchs am Ende der Eingabe
    let name = name.trim();

    // Begrüßung mit dem Namen
    println!("Hallo, {}! Schön, dass Sie hier sind.", name);

    // Programm beenden
    println!("Das Programm wurde beendet.");
}
